src/sql/workbench/contrib/notebook/test/electron-browser/notebookUtils.test.ts

```diff
@@ -3,86 +3,184 @@
-		let capabilitiesService = new TestCapabilitiesService();
-		let connProfile = new ConnectionProfile(capabilitiesService, conn);
-		connProfile.serverName = 'serverName';
-		let attachToNameFormatted = formatServerNameWithDatabaseNameForAttachTo(connProfile);
-		assert.equal(attachToNameFormatted, 'serverName');
+	test('getProvidersForFileName Test', async function (): Promise<void> {
+		setupMockNotebookService();
+
+		let result = getProvidersForFileName('', mockNotebookService.object);
+		assert.deepStrictEqual(result, [defaultTestProvider]);
+
+		result = getProvidersForFileName('fileWithoutExtension', mockNotebookService.object);
+		assert.deepStrictEqual(result, [defaultTestProvider]);
+
+		result = getProvidersForFileName('test.sql', mockNotebookService.object);
+		assert.deepStrictEqual(result, [testProvider]);
+
+		mockNotebookService.setup(n => n.getProvidersForFileType(TypeMoq.It.isAnyString()))
+			.returns(() => undefined);
+		result = getProvidersForFileName('test.sql', mockNotebookService.object);
+		assert.deepStrictEqual(result, [DEFAULT_NOTEBOOK_PROVIDER]);
 	});
 
-	test('Should format server name correctly for attach to when database is undefined', async function (): Promise<void> {
-		let capabilitiesService = new TestCapabilitiesService();
-		let connProfile = new ConnectionProfile(capabilitiesService, conn);
-		connProfile.serverName = 'serverName';
-		connProfile.databaseName = undefined;
-		let attachToNameFormatted = formatServerNameWithDatabaseNameForAttachTo(connProfile);
-		assert.equal(attachToNameFormatted, 'serverName');
+	test('getStandardKernelsForProvider Test', async function (): Promise<void> {
+		setupMockNotebookService();
+
+		let result = getStandardKernelsForProvider(undefined, undefined);
+		assert.deepStrictEqual(result, []);
+
+		result = getStandardKernelsForProvider(undefined, mockNotebookService.object);
+		assert.deepStrictEqual(result, []);
+
+		result = getStandardKernelsForProvider('testProvider', undefined);
+		assert.deepStrictEqual(result, []);
+
+		result = getStandardKernelsForProvider('testProvider', mockNotebookService.object);
+		assert.deepStrictEqual(result, [<IStandardKernelWithProvider>{
+			name: 'testName',
+			displayName: 'testDisplayName',
+			connectionProviderIds: ['testId1', 'testId2'],
+			notebookProvider: 'testProvider'
+		}]);
 	});
 
-	test('Should format server name as empty string when server/database are undefined', async function (): Promise<void> {
-		let capabilitiesService = new TestCapabilitiesService();
-		let connProfile = new ConnectionProfile(capabilitiesService, conn);
-		connProfile.serverName = undefined;
-		connProfile.databaseName = undefined;
-		let attachToNameFormatted = formatServerNameWithDatabaseNameForAttachTo(connProfile);
-		assert.equal(attachToNameFormatted, '');
+	test('tryMatchCellMagic Test', async function (): Promise<void> {
+		let result = tryMatchCellMagic(undefined);
```

Comment:
how about a couple of tests with multiline content?

Reply:
Done